src/databricks/labs/remorph/deployment/configurator.py

```diff
@@ -146,3 +99,113 @@ def warehouse_type(_):
             )
             warehouse_id = new_warehouse.id
         return warehouse_id
+
+    def has_necessary_catalog_access(
+        self, catalog_name: str, user_name: str, privilege_sets: tuple[set[Privilege], ...]
+    ):
+        catalog = self._catalog_ops.get_catalog(catalog_name)
+        assert catalog
+        if self._catalog_ops.has_catalog_access(catalog, user_name, privilege_sets):
+            return True
+        privilege_set_permissions = [
+            (
+                privilege_set,
+                self._catalog_ops.has_privileges(user_name, SecurableType.CATALOG, catalog.name, privilege_set),
+            )
+            for privilege_set in privilege_sets
+        ]
+
+        missing_permissions = ResourceConfigurator._get_missing_permissions(privilege_set_permissions)
+
+        logger.error(
+            f"User `{user_name}` doesn't have required privileges `{missing_permissions}` \n to access catalog `{catalog_name}` "
+        )
+        return False
+
+    def has_necessary_schema_access(
+        self, catalog_name: str, schema_name: str, user_name: str, privilege_sets: tuple[set[Privilege], ...]
+    ):
+        schema = self._catalog_ops.get_schema(catalog_name, schema_name)
+        assert schema
```

Comment:
better change catalog ops to return SchemaInfo or throw NotFound exception

Reply:
It would impact existing mocked Test cases. So, added the missing assertion message